integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -0,0 +1,1361 @@
+          const { subs, query } = addSubscriptionForPerson(this.realm);
+
+          subs.update((mutableSubs) => {
+            mutableSubs.remove(query);
+            mutableSubs.add(this.realm.objects(FlexiblePersonSchema.name).filtered("age < 10"));
+            mutableSubs.add(this.realm.objects(FlexiblePersonSchema.name).filtered("age > 20"));
+            mutableSubs.add(this.realm.objects(DogSchema.name).filtered("age > 30"));
+          });
+
+          expect(subs).to.have.length(3);
+
+          expect(subs[0].queryString).to.equal("age < 10");
+          expect(subs[0].objectType).to.equal(FlexiblePersonSchema.name);
+
+          expect(subs[1].queryString).to.equal("age > 20");
+          expect(subs[1].objectType).to.equal(FlexiblePersonSchema.name);
+
+          expect(subs[2].queryString).to.equal("age > 30");
+          expect(subs[2].objectType).to.equal(DogSchema.name);
+        });
+
+        it("handles multiple updates in multiple batches", function (this: RealmContext) {
+          const { subs, query } = addSubscriptionForPerson(this.realm);
+
+          subs.update((mutableSubs) => {
+            mutableSubs.remove(query);
+            mutableSubs.add(this.realm.objects(FlexiblePersonSchema.name).filtered("age < 10"));
+          });
+
+          subs.update((mutableSubs) => {
+            mutableSubs.add(this.realm.objects(FlexiblePersonSchema.name).filtered("age > 20"));
+            mutableSubs.add(this.realm.objects(DogSchema.name).filtered("age > 30"));
+          });
+
+          expect(subs).to.have.length(3);
+
+          expect(subs[0].queryString).to.equal("age < 10");
+          expect(subs[0].objectType).to.equal(FlexiblePersonSchema.name);
+
+          expect(subs[1].queryString).to.equal("age > 20");
+          expect(subs[1].objectType).to.equal(FlexiblePersonSchema.name);
+
+          expect(subs[2].queryString).to.equal("age > 30");
+          expect(subs[2].objectType).to.equal(DogSchema.name);
+        });
+
+        it("still applies all updates in a batch if one errors", async function (this: RealmContext) {
+          const { subs } = addSubscriptionForPerson(this.realm);
+
+          try {
+            subs.update((mutableSubs) => {
+              mutableSubs.add(this.realm.objects(FlexiblePersonSchema.name).filtered("age < 10"));
+              mutableSubs.add(this.realm.objects(FlexiblePersonSchema.name).filtered("nonQueryable == 'test'"));
+              mutableSubs.add(this.realm.objects(DogSchema.name).filtered("age > 30"));
+            });
+          } catch (e) {
+            // Error is expected as `nonQueryable` field is not queryable
+          }
+
+          await expect(subs.waitForSynchronization()).to.be.rejected;
+          expect(subs.state).to.equal(Realm.App.Sync.SubscriptionsState.Error);
+
+          expect(subs).to.have.length(4);
+        });
+
+        it("allows an empty update", async function () {
+          const { subs } = addSubscriptionForPerson(this.realm);
```

Comment:
My intention in general was to have something "interesting" to assert about the state of the Realm – i.e. "it had one object to start with and it still has one object" seems a little more "interesting" than "it had no objects and it still has no objects", because we know nothing has been deleted or added (unless something got both deleted and added!). Probably more applicable to the `remove` functions but I thought it seemed reasonable to do elsewhere, happy to either remove or comment if you think otherwise